sdk/src/main/java/com/hedera/hashgraph/sdk/Pem.java

```diff
@@ -122,18 +117,27 @@ static void writeEncryptedPrivateKey(PrivateKeyInfo pkInfo, Writer out, String p
      */
     static PrivateKeyInfo readPrivateKey(Reader input, @Nullable String passphrase) throws IOException {
         try {
-            var decryptProvider = new JceOpenSSLPKCS8DecryptorProviderBuilder()
-                .setProvider(new BouncyCastleProvider())
-                .build(passphrase != null ? passphrase.toCharArray() : "".toCharArray());
             try (final var parser = new PEMParser(input)) {
                 Object parsedObject = parser.readObject();
                 if (parsedObject == null) {
                     throw new BadKeyException("PEM file did not contain a private key");
                 } else if (parsedObject instanceof PKCS8EncryptedPrivateKeyInfo) {
+                    var decryptProvider = new JceOpenSSLPKCS8DecryptorProviderBuilder()
+                        .setProvider(new BouncyCastleProvider())
+                        .build(passphrase != null ? passphrase.toCharArray() : "".toCharArray());
```

Comment:
I guess we can extract that part for readability instead of every if doing a ternary expression.